global_helpers/panther_greynoise_helpers.py

```diff
@@ -84,15 +84,52 @@ def vpn_service(self, match_field) -> str:
         return deep_get(self.noise, match_field, "vpn_service")
 
 
-class GreyNoiseRIOT:
+class GreyNoiseRIOTBasic:
     def __init__(self, event):
         self.riot = deep_get(event, "p_enrichment", "greynoise_riot_basic")
 
-    def ip_range(self, match_field) -> str:
-        return deep_get(self.riot, match_field, "ip_cidr")
+    def ip_address(self, match_field) -> str:
+        return deep_get(self.riot, match_field, "provider", "ip")
+
+    def classification(self, match_field) -> str:
+        return deep_get(self.riot, match_field, "provider", "classification")
+
+    def name(self, match_field) -> str:
+        return deep_get(self.riot, match_field, "provider", "name")
+
+    def url(self, match_field) -> str:
+        ip_addr = deep_get(self.riot, match_field, "provider", "ip")
+        return f"https://viz.greynoise.io/riot/{ip_addr}"
+
+    def last_seen(self, match_field) -> datetime.date:
+        return parser.parse(deep_get(self.riot, match_field, "provider", "last_seen"))
+
+
+class GreyNoiseRIOTAdvanced:
+    def __init__(self, event):
+        self.riot = deep_get(event, "p_enrichment", "greynoise_riot_advanced")
+
+    def ip_address(self, match_field) -> str:
+        return deep_get(self.riot, match_field, "provider", "ip")
+
+    def classification(self, match_field) -> str:
+        return deep_get(self.riot, match_field, "provider", "classification")
+
+    def name(self, match_field) -> str:
```

Comment:
This is a duplicate, not sure if we need it in both?

Reply:
Yeah these classes could probably inherit from one another but I wanted to add some logic in the __init__ functions to ensure the user is using the correct helper for their subscription level, and raise an error if not. Still probably good do some inheritance but this felt cleaner